.github/workflows/CI-tag-orchestrator.yml

```diff
@@ -5,54 +5,132 @@ on:
+      artifact_name_relay: ${{ steps.create-tar-file.outputs.tar_file_name_relay }}
+
     steps:
-      - name: Git checkout
-        uses: actions/checkout@v4
+      - uses: actions/checkout@v4
         with:
-          ref: ${{ github.ref }}
+          ref: ${{ github.event.inputs.git_tag || github.ref }}
           fetch-depth: 0
 
-      - name: Download zkVerify Docker image artifact
-        uses: actions/download-artifact@v4
+      - name: Set up deps cache
+        run: mkdir deps
+
+      - name: Restore common cache
+        if: ${{ !env.ACT }}
+        uses: actions/cache/restore@v3
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            deps/
+          key: common-cache
+      
+      - name: Restore build cache
+        if: ${{ !env.ACT }}
+        uses: actions/cache/restore@v3
         with:
-          name: zkverify-image
-          path: ./
+          path: |
+            target/
+          key: build-cache
       
-      - name: Build/Publish Docker image
-        env:
-          RELEASE_BRANCH: release
-          MAINTAINERS_KEYS: ${{ vars.MAINTAINERS_KEYS }}
-          DOCKER_HUB_USERNAME: ${{ secrets.DOCKER_HUB_USERNAME }}
-          DOCKER_HUB_TOKEN: ${{ secrets.DOCKER_HUB_TOKEN }}
+      - name: Verify release build
+        run: |
+          ls target
+          ls target/release
+
+      - name: Build solo and relay chain configured images
+        id: build_images
+        run: |
+          SKIP_IMAGE_VALIDATION=true
+          "${GITHUB_WORKSPACE}/docker/scripts/bootstrap.sh" --skip-parachain --skip-compile
+        shell: bash
+
+      - name: Set Docker image names as output
+        id: save-image-names
         run: |
-          source "${GITHUB_WORKSPACE}/ci/setup_env.sh"
-          "${GITHUB_WORKSPACE}/ci/docker.sh" --image-artifact zkverify-image
+          echo "docker_image_solo=horizenlabs/zkverify:latest" >> $GITHUB_OUTPUT
+          echo "docker_image_relay=horizenlabs/zkv-relay:latest" >> $GITHUB_OUTPUT
+
+      - name: Save the Docker image as a tarball
+        id: create-tar-file
+        run: |           
+          TAR_FILE_NAME_SOLO="zkverify-solo"
+          TAR_FILE_SOLO="${{github.workspace}}/${TAR_FILE_NAME_SOLO}.tar"
+
+          TAR_FILE_NAME_RELAY="zkverify-relay"
+          TAR_FILE_RELAY="${{github.workspace}}/${TAR_FILE_NAME_RELAY}.tar"
+
+          echo "tar_file_name_solo=$TAR_FILE_NAME_SOLO" >> $GITHUB_OUTPUT
+          echo "tar_file_solo=$TAR_FILE_SOLO" >> $GITHUB_OUTPUT
+
+          echo "tar_file_name_relay=$TAR_FILE_NAME_RELAY" >> $GITHUB_OUTPUT
+          echo "tar_file_relay=$TAR_FILE_RELAY" >> $GITHUB_OUTPUT
+
+          docker save "${{ steps.save-image-names.outputs.docker_image_solo }}" -o "${TAR_FILE_SOLO}"
+          docker save "${{ steps.save-image-names.outputs.docker_image_relay }}" -o "${TAR_FILE_RELAY}"
```

Comment:
Why do we need `save-image-names` step if we can just use the values we are saving directly here:
`docker save horizenlabs/zkverify:latest -o ....` :thinking: